integration/test/ca_bundle/ca_bundle_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+// +build linux
+// +build integration
+
+package ca_bundle
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+const configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const commonConfigOutputPath = "/opt/aws/amazon-cloudwatch-agent/etc/common-config.toml"
+const configJSON = "/config.json"
+const commonConfigTOML = "/common-config.toml"
+const outputLog = "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log"
+const targetString = "x509: certificate signed by unknown authority"
+//Let the agent run for 1 minutes. This will give agent enough time to call server
+const agentRuntime = 1 * time.Minute
+
+
+type input struct {
+	findTarget bool
+	dataInput string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+//This test uses a pem file created for the local stack endpoint to be able to coneect via ssl
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		//Use the system pem ca bundle  + local stack pem file ssl should connect thus target string not found
+		{dataInput: "resources/integration/ssl/with/combine/bundle", findTarget: false},
+		//Use the system pem ca bundle ssl should not connect thus target string found
+		{dataInput: "resources/integration/ssl/with/original/bundle", findTarget: true},
+		//Do not look for ca bundle should not connect thus target string found
+		{dataInput: "resources/integration/ssl/without/bundle", findTarget: true},
+	}
+
+	for _, parameter := range parameters {
+		//before test run
+		log.Printf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget)
+		clearLogFile()
+		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget), func(t *testing.T) {
+			copyFile(parameter.dataInput + configJSON, configOutputPath)
+			copyFile(parameter.dataInput + commonConfigTOML, commonConfigOutputPath)
+			startAgent();
+			time.Sleep(agentRuntime);
+			log.Printf("Agent has been running for : %s", agentRuntime.String());
+			stopAgent();
+			containsTarget := readOutputLog()
+			if (parameter.findTarget && !containsTarget) || (!parameter.findTarget && containsTarget) {
+				t.Errorf("Find target is %t contains target is %t", parameter.findTarget, containsTarget)
+			}
+		})
+	}
+}
+
+func clearLogFile() {
+	cmd := exec.Command("bash", "-c", "sudo rm " + outputLog + " && sudo touch " + outputLog)
```

Comment:
You can configure the agent to print to stderr and use a buffer to capture all of that instead of rotating log files.
> `logfile` – Specifies the location where the CloudWatch agent writes log messages. If you specify an empty string, the log goes to stderr.

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html#CloudWatch-Agent-Configuration-File-Agentsection

```
cmd := exec.Command(bin, cmdArgs...)
var buf bytes.Buffer
cmd.Stderr = &buf
```